File: 4/48/RotateImage.java
/*
https://leetcode.com/problems/rotate-image/description/?envType=daily-question&envId=2026-05-04
*/

class Solution {
    public void rotate(int[][] matrix) {
        int rows=matrix.length, columns=matrix[0].length;
        int[][]resultantMatrix=new int[rows][columns];
        int colTrvStartIdx=columns-1;
        for(int i=0; i<rows; ++i){
            int[] currentRow=matrix[i];
            int toTraverseIndex=0;
            for(int ii=0; ii<rows; ++ii){
                resultantMatrix[ii][colTrvStartIdx]=currentRow[toTraverseIndex++];
            }
            colTrvStartIdx--;
        }
        for(int i=0; i<rows; ++i){
            for(int j=0; j<columns; ++j){
                matrix[i][j]=resultantMatrix[i][j];
            }
        }
    }
}

File: 4/48/RotateImage.kt
/*
https://leetcode.com/problems/rotate-image/description/?envType=daily-question&envId=2026-05-04
*/

class Solution {
    fun rotate(matrix: Array<IntArray>) {
        val rows:Int=matrix.size
        val cols:Int=matrix[0].size
        val resultantMatrix=mutableListOf<IntArray>()
        var colTrvStartIdx:Int=cols-1
        for(i in 0 until rows){
            val newRow=IntArray(cols)
            resultantMatrix.add(newRow)
        }
        for(i in 0 until rows){
            val currentRow:IntArray=matrix[i]
            var toTraverseIndex:Int=0
            for(ii in 0 until rows){
                resultantMatrix[ii][colTrvStartIdx]=currentRow[toTraverseIndex++]
            }
            colTrvStartIdx--
        }
        for(i in 0 until rows){
            for(j in 0 until cols){
                matrix[i][j]=resultantMatrix[i][j]
            }
        }
    }
}